File: tests/fixtures/check_then_act/py/finally_after_return_fixture.py
# A try body that returns still runs its finally, so a write there comes after the await.
async def finish(repo, item_id):
    row = repo.get(item_id)
    if row:  # expect: check
        return
    try:
        await repo.sync(item_id)  # expect: gap await
        return
    finally:
        repo.update(item_id)  # expect: write

File: tests/fixtures/check_then_act/ts/finally_after_return_fixture.ts
// A try block that returns still runs its finally, so a write there comes after the await.
export async function finish(repo: any, id: string) {
  const row = repo.get(id);
  if (row) return; // expect: check
  try {
    await repo.sync(id); // expect: gap await
    return;
  } finally {
    repo.update(id); // expect: write
  }
}
